### DISCUSSION TITLE
issue: defalutValues not behaving as expected

### DISCUSSION THREAD

**Discussion by janusqa** [react-hook-form/react-hook-form#10404] [Q&A]
### Version Number

7.43.9

### Codesandbox/Expo snack

none

### Steps to reproduce

I have a form where validation is like 
```ts
const FormInput = z.object({
    title: z.string({ required_error: 'Title is required' }),
    price: z.coerce
        .number({
            required_error: 'Price is required',
            invalid_type_error: 'Price field is a required',
        })
        .min(1, { message: 'Price must be greater than 0' })
        .max(10000, { message: 'Price must be less than 10000' }),
    // price: z.preprocess((arg) => {
    //     const processed = z.string().transform(Number).safeParse(arg);
    //     return processed.success ? processed.data : arg;
    // }, z.number({ required_error: 'Price is required', invalid_type_error: 'Price field is a required' }).min(1, { message: 'Price must be greater than 0' }).max(10000, { message: 'Price must be less than 10000' })),
    description: z.string().optional(),
    category: z.object({
        label: z.enum(categories),
        value: z.number(),
        color: z.string().optional(),
        icon: z.string().optional(),
    }),
    images: z.array(z.string()).nonempty({ message: 'Please select an image' }),
});

const defaultValues = {
    title: '',
    price: '',
    description: '',
    category: { label: '', value: 1, color: '', icon: '' },
    images: [] as string[],
};
```

I can submit this form with an empty title although I have specified in zod scheme that title is a required field
typscript flags default values as an error because price is "" instead of a number. 

### Expected behaviour

Form should not be allowed to be submitted with and empty string 
Should be able to specify default value of "" for number

The reason is when I use reset i want these fields to be empty.
If i do not put title as "" in default values then this field is not reset when using the reset function
If I put price as 0 then its reset to 0 instead of making the field empty.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RMuradli** [comment]
My personal advice would be to use the validator of React-hook-form. After using yup schema too many times, I have learned that schema object gets less and less readable as the complexity of the form increases. Plus you have to consider everything very carefully, like triggering for example, will not work the same with a schema object as it does with factory default
